chainerio/__init__.py

```diff
@@ -68,60 +87,194 @@ def open(file_path: str, mode: str = 'rb',
         errors, newline, closefd, opener)
 
 
-def set_root(uri_or_handler: str) -> None:
+def set_root(uri_or_handler: Union[str, Type['IO']]) -> None:
+    """Sets the current context to ``uri_or_handler``
+
+    The context here refers to the default handler and the root path.
+    The default handler points to a filesystem or a container which
+    ChainerIO uses when called without a full URI. Overrides by the full URI.
+    The default handler can be created by :func:`create_handler` with
+    the name of the handler.
+
+    Example::
+
+        # Case 1
+        # set_root by the name of uri
+        chainerio.set_root("posix")
+        # open a file on posix filesystem with path "some/file"
+        chainerio.open("some/file")
+        # override with a full uri
+        chainerio.open("hdfs:///some/file/on/hdfs")
+
+        # Case 2
+        # set_root by uri
+        chainerio.set_root("hdfs:///some/directory")
+        # open a file "on/hdfs" on hdfs under the " some/directory"
+        chainerio.open("on/hdfs")
+
+        # Case 3
+        handler = chainerio.create_handler('hdfs')
+        # set_root by handler
+        chainerio.set_root(handler)
+        # open a file "/some/file/on/hdfs" on hdfs
+        chainerio.open("/some/file/on/hdfs")
+
+        # Case 4
+        handler = chainerio.open_as_container('some.zip')
+        # set_root by handler
+        chainerio.set_root(handler)
+        # open a file "img.jpg" in "some.zip"
+        chainerio.open("img.jpg")
+
+    The root path refers to a directory that ChainerIO works on.
+    It is similar to current working directory, ``CWD`` in terms of the shell
+    environment.
+    The root path will only be set
+    when the ``uri_or_handler`` points to a directory.
+    Otherwise, it will be set to default,
+    which represents the default working directory as follow:
+
+    +---------+---------------------------+
+    |         |     Default Root Path     |
+    +=========+===========================+
+    | POSIX   | current working directory |
+    +---------+---------------------------+
+    | HDFS    |     /user/``USERNAME`` #1 |
+    +---------+---------------------------+
+    | zip     |      top directory        |
+    +---------+---------------------------+
+
+    #1 For the details about the ``USERNAME`` in HDFS, please refer to
+    `HDFS Document <https://hadoop.apache.org/docs/current/\
+            hadoop-project-dist/hadoop-hdfs/HdfsPermissionsGuide.html>`_
+
+
+    Args:
+        uri_or_handler (str or :class:~`chainerio.IO`): The ``uri_or_handler``
+            can accept the following three kinds of values:
+
+            1. the name of a handler (string): set the default handler
```

Comment:
Technically the name of a handler isn't clear through the document IMO. Would be better if it's "scheme" here